pandas/core/reshape/merge.py

```diff
@@ -2399,133 +2374,27 @@ def _factorize_keys(
     if (
         isinstance(lk.dtype, DatetimeTZDtype) and isinstance(rk.dtype, DatetimeTZDtype)
     ) or (lib.is_np_dtype(lk.dtype, "M") and lib.is_np_dtype(rk.dtype, "M")):
-        # Extract the ndarray (UTC-localized) values
         # Note: we dont need the dtypes to match, as these can still be compared
         lk, rk = cast("DatetimeArray", lk)._ensure_matching_resos(rk)
```

Comment:
should the ensure_matching_resos be done in the EA method similar to how the Categorical one does encode_with_whatever?

Reply:
i see, never mind

Reply:
do we have tests with timedelta64 and mismatched resos?

Reply:
Don't know